### PR TITLE
Add setup instructions for GA and GTM to README

### PR DESCRIPTION
**Pull request by Gayathri625**
Updated README.md to include setup and verification instructions for Google Analytics and Google Tag Manager.

### CODE REVIEW

**Review by arjan-bal — CHANGES_REQUESTED** [review]
Please also mention building with the PRODUCTION variable so that GA works.

**Comment by gnossen** [comment]
Seems like https://github.com/grpc/grpc.io/pull/1332#pullrequestreview-2197470755 still needs to be resolved, so holding off on merging for now.

**Comment by arjan-bal** [comment]
> Please also mention building with the PRODUCTION variable so that GA works.

@Gayathri625  For reference, this seems to be the block that only includes analytics scripts when in production mode: https://github.com/grpc/grpc.io-docsy/blob/ba20af1c9044e3ad8d83ead5153b243c88539038/layouts/partials/head.html#L55-L61

**Comment by Gayathri625** [comment]
> I see that we load GTM scripts in non-prod also in #1313. Can you have a similar prod env check for them also? You can do that in a separate PR.

check for production environment: [#1335](https://github.com/grpc/grpc.io/pull/1335)

File: README.md [hunk 1]
```diff
@@ -39,6 +39,33 @@ To locally serve the site at [localhost:8888][], run the following command:
...
+   - Follow the [GTM setup guide](https://developers.google.com/tag-platform/tag-manager/web) to create your account and container.
+   - Update `config.yaml` with your GTM ID:
+
+   ```yaml
+   params:
+     gtmID: "GTM-XXXXXXXX" # Replace with your GTM ID
+   ```
+    Check GTM setup using Preview mode and verify that Tag Assistant displays "Connected" for your website URL.
+   
+Build the site by setting Hugo environment to production to make sure GA and GTM work correctly
```
Comment (arjan-bal) [inline README.md:64]: nit: Can omit "correctly", they will not work at all without it,
>...make sure GA and GTM work.

**Review by arjan-bal — APPROVED** [review]
LGTM!

**Review by gnossen — APPROVED** [review]

**Pull request merged**